Allow layout builder blocks to be translated

pfrilling opened the pull request:
## Summary
This updates the custom block types to be translatable. This also updates all custom paragraph entities to be translatable. I re-exported the features with the most recent Covid database to capture any overrides that were showing.

A `hook_preprocess_block` was added to allow the custom and inline block labels to be translated.

## Metadata
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | no
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | no
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | [RIG-160](https://thinkoomph.jira.com/browse/rig-160)

ben-hamelin commented on ecms_base/features/custom/ecms_person/config/install/core.base_field_override.node.person.path.yml:
```diff
@@ -0,0 +1,19 @@
+label: 'URL alias'
+description: ''
+required: false
+translatable: true
```
Should path be translatable "false" ?

ben-hamelin commented on ecms_base/features/custom/ecms_press_release/config/install/core.base_field_override.node.press_release.path.yml:
```diff
@@ -0,0 +1,19 @@
+label: 'URL alias'
+description: ''
+required: false
+translatable: true
```
Should path be false?

ben-hamelin commented on ecms_base/features/custom/ecms_promotions/config/install/core.base_field_override.node.promotions.path.yml:
```diff
@@ -0,0 +1,19 @@
+label: 'URL alias'
+description: ''
+required: false
+translatable: true
```
Should this be false?

ben-hamelin commented on ecms_base/themes/custom/ecms/ecms.theme:
```diff
@@ -86,7 +86,7 @@ function ecms_preprocess_paragraph__media_item(array &$variables): void {
 
 /**
- * Implements hook_preprocess_paragaph().
+ * Implements hook_preprocess_paragraph().
```
nitpick - this doesn't match the hook_

ben-hamelin commented on ecms_base/themes/custom/ecms/ecms.theme:
```diff
@@ -253,3 +253,17 @@ function ecms_form_alter(&$form, FormStateInterface $form_state, string $form_id
+/**
+ * Implements hook_preprocess_block().
+ */
+function ecms_preprocess_block(&$variables): void {
```
argument typ

ben-hamelin commented on ecms_base/features/custom/ecms_landing_page/config/install/core.base_field_override.block_content.content_components.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: block_content
+bundle: content_components
+label: Metatags
+description: 'The meta tags for the entity.'
```
I don't think we want meta tags on a block type

ben-hamelin commented on ecms_base/features/custom/ecms_landing_page/config/install/core.base_field_override.block_content.page_title_with_photo.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: block_content
+bundle: page_title_with_photo
+label: Metatags
+description: 'The meta tags for the entity.'
```
Should remove this field

ben-hamelin commented on ecms_base/features/custom/ecms_landing_page/config/install/core.base_field_override.block_content.text_card_collection.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: block_content
+bundle: text_card_collection
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.file.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: media
+bundle: file
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_audio.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: media
+bundle: media_item_audio
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_image.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: media
+bundle: media_item_image
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_video.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: media
+bundle: media_item_video
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_promotions/config/install/core.base_field_override.block_content.promotion_reference.metatag.yml:
```diff
@@ -0,0 +1,17 @@
+entity_type: block_content
+bundle: promotion_reference
+label: Metatags
+description: 'The meta tags for the entity.'
```
Remove metatag field

ben-hamelin commented on ecms_base/features/custom/ecms_promotions/config/install/core.base_field_override.media.promotional_image.status.yml:
```diff
@@ -0,0 +1,21 @@
+label: Published
+description: ''
+required: false
+translatable: false
```
Should this be true to allow for unpublishing in different languages?

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.file.status.yml:
```diff
@@ -0,0 +1,21 @@
+label: Published
+description: ''
+required: false
+translatable: false
```
Should this be true?

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_audio.status.yml:
```diff
@@ -0,0 +1,21 @@
+label: Published
+description: ''
+required: false
+translatable: false
```
Should this be true?

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_image.status.yml:
```diff
@@ -0,0 +1,21 @@
+label: Published
+description: ''
+required: false
+translatable: false
```
Should this be true?

ben-hamelin commented on ecms_base/features/custom/ecms_paragraphs/config/install/core.base_field_override.media.media_item_video.status.yml:
```diff
@@ -0,0 +1,21 @@
+label: Published
+description: ''
+required: false
+translatable: false
```
Should this be true?

ben-hamelin reviewed (changes requested):
Have a few questions on the need for metatag fields on blocks and paragraphs, and the translation setting for "status" fields.

pfrilling commented:
@ben-hamelin, I fixed the translation configurations you found and updated the php in the .theme file. As for the metatag overrides. Those all came from exporting out of the covid site. I don't think it will hurt anything having them in the codebase.

With these configuration updates, `features:list` give us a clean features list:

```
Philips-MacBook-Pro-2:develop-ecms-profile pfrilling$ ldr fl --bundle=ecms
 ----------------------- ----------------------- -------------- --------- -------
  Name                    Machine name            Status         Version   State
 ----------------------- ----------------------- -------------- --------- -------
  Basic page              ecms_basic_page         Installed      9.x-1.0
  Event                   ecms_event              Installed      9.x-1.0
  eCMS Executive Orders   ecms_executive_orders   Uninstalled    9.x-1.0
  eCMS Hotels             ecms_hotel              Uninstalled    9.x-1.0
  Landing page            ecms_landing_page       Installed      9.x-1.0
  Location                ecms_location           Installed      9.x-1.0
  Notification            ecms_notification       Installed      9.x-1.0
  eCMS Paragraphs         ecms_paragraphs         Installed      9.x-1.0
  Person                  ecms_person             Installed      9.x-1.0
  Press release           ecms_press_release      Installed      9.x-1.0
  eCMS Projects           ecms_projects           Uninstalled    9.x-1.0
  Promotions              ecms_promotions         Installed      9.x-1.0
  eCMS Publications       ecms_publications       Uninstalled    9.x-1.0
  eCMS Speeches           ecms_speeches           Uninstalled    9.x-1.0
  Webform                 webform                 Not exported
  User                    user                    Not exported
  Core                    core                    Not exported
  Site                    site                    Not exported
 ----------------------- ----------------------- -------------- --------- -------
```

ben-hamelin commented:
Ok looks good!
/approved
